samtranslator/swagger/swagger.py

```diff
@@ -695,6 +700,10 @@ def add_auth_to_method(self, path: str, method_name: str, auth: Dict[str, Any],
             self._set_method_authorizer(path, method_name, method_authorizer, authorizers, method_scopes)  # type: ignore[no-untyped-call]
 
         method_apikey_required = auth and auth.get("ApiKeyRequired")
+
+        if not auth.get("AddApiKeyRequiredToCorsPreflight") and method_name == "options":
```

Comment:
So it's symmetrical (and easier to read) with the other very similar check:

```suggestion
        if method_name == "options" and not auth.get("AddApiKeyRequiredToCorsPreflight"):
```